src/saveToCase.js

```diff
@@ -0,0 +1,495 @@
+define([
+    'vellum/form',
+    'jquery',
+    'underscore',
+    'vellum/mugs',
+    'vellum/parser',
+    'vellum/tree',
+    'vellum/util',
+    'vellum/widgets',
+    'tpl!vellum/templates/widget_update_case',
+    'vellum/core'
+], function (
+    form_,
+    $,
+    _,
+    mugs,
+    parser,
+    Tree,
+    util,
+    widgets,
+    widget_update_case
+){
+    function createsCase(mug) {
+        return mug ? mug.p.use_create : false;
+    }
+
+    function closesCase(mug) {
+        return mug ? mug.p.use_close : false;
+    }
+
+    function updatesCase(mug) {
+        return mug ? mug.p.use_update : false;
+    }
+
+    function addSetValue(mug) {
+        var path = mug.absolutePath;
+
+        if (createsCase(mug)) {
+            mug.form.addSetValue('xforms-ready', path + "/case/@case_id", mug.p.case_id);
+        }
+    }
+
+    var saveCasePropWidget = function (mug, options) {
```

Comment:
Close to https://github.com/dimagi/Vellum/blob/master/src/widgets.js#L228 but based on `normal` instead of `base`